cloud/services/domains.go

```diff
@@ -97,37 +103,38 @@ func DeleteIPFromDNS(ctx context.Context, mscope *scope.MachineScope) error {
 	return nil
 }
 
-// GetMachinePublicIP gets the machines public IP
-func GetMachinePublicIP(ctx context.Context, mscope *scope.MachineScope) (string, error) {
+// GetMachinePublicIPs gets the machines public IP
+func GetMachinePublicIPs(ctx context.Context, mscope *scope.MachineScope) (map[string]string, error) {
 	// Verify instance id is not nil
 	if mscope.LinodeMachine.Spec.InstanceID == nil {
 		err := errors.New("instance ID is nil. cant get machine's public ip")
-		return "", err
+		return nil, err
 	}
 
 	// Get the public IP that was assigned
 	addresses, err := mscope.LinodeClient.GetInstanceIPAddresses(ctx, *mscope.LinodeMachine.Spec.InstanceID)
 	if err != nil {
-		return "", fmt.Errorf("failed to get ip address of the instance: %w", err)
+		return nil, err
 	}
-	if len(addresses.IPv4.Public) == 0 {
-		err := errors.New("no public IP address")
-		return "", err
+
+	if len(addresses.IPv4.Public) == 0 || addresses.IPv6 == nil {
+		err := errors.New("no public address")
+		return nil, err
 	}
 
-	return addresses.IPv4.Public[0].Address, nil
+	return map[string]string{"IPv4": addresses.IPv4.Public[0].Address, "IPv6": addresses.IPv6.SLAAC.Address}, nil
```

Comment:
We may want to check `addresses.IPV6.SLAAC` isn't nil

Reply:
Added